src/main/java/com/metamx/common/parsers/Parser.java

```diff
@@ -20,9 +20,30 @@
 import java.util.Map;
 import java.util.Set;
 
+/**
+ * Class that can parse Strings into Maps.
+ */
 public interface Parser<K, V>
 {
-  public Map<K, V> parse(String input) ;
-  public void setFieldNames(Iterable<String> fieldNames) ;
+  /**
+   * Parse a String into a Map.
+   *
+   * @throws ParseException if the String cannot be parsed
+   */
+  public Map<K, V> parse(String input);
+
+  /**
+   * Set the fieldNames that you expect to see in parsed Maps. Deprecated; Parsers should not, in general, be
+   * expected to know what fields they will return. Some individual types of parsers do need to know (like a TSV
+   * parser) and those parsers have their own way of setting field names.
+   */
+  @Deprecated
+  public void setFieldNames(Iterable<String> fieldNames);
+
+  /**
+   * Returns the fieldNames that we expect to see in parsed Maps, if known, or null otherwise. Deprecated; Parsers
+   * should not, in general, be expected to know what fields they will return.
+   */
+  @Deprecated
   public List<String> getFieldNames();
```

Comment:
@himanshug the thought behind deprecating these was:

- external usages of setFieldNames are generally immediately after construction of a specific parser class. So those usages don't really need to use a setter- they could instead use a constructor that offers fieldNames (those constructors already exist but are not always used).

- these methods are actually pretty nonsensical for the JsonPath parser (#34), because its FieldSpecs serve the same purpose, but they do it in a richer way that is specific to JsonPath.

So, I don't think these methods are really useful, nor do they generalize well to all parsers.